### PR TITLE
fix(origin): confirm paths redirect to subdomain

### PR DESCRIPTION
**Pull request by lidel**
This PR makes the Origin check more strict: after confirming subdomains work, additional request for path is made, to confirm legacy paths are redirected to expected subdomains.

Adding this test is necessary, because there are misconfigured gateways that don't redirect /ipfs/{cid} paths to {cid}.ipfs.example.com  subdomains and return data from the root origin, effectively defeating any Origin isolation provided by subdomains. :see_no_evil: 

This fix will incentivize gateway operators to fix such misconfigurations + inform users about real risks.

cc @schomatis fyi an example of misconfigured gateway: 

- subdomain "works" (is naively redirected to dweb.link for some reason): https://bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m.ipfs.ipfs.drink.cafe/#x-ipfs-companion-no-redirect
- ..but is defeated by the lack of redirect: https://ipfs.drink.cafe/ipfs/bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m#x-ipfs-companion-no-redirect

**TLDR**: This PR correctly shows ⚠️ next to `ipfs.drink.cafe` :)

### CODE REVIEW

**Comment by schomatis** [comment]
I can't review this code but the check you're describing makes perfect sense to me, thanks. My only request would be that CID we're using to test subdomain redirection not be static so a malicious/lazy gateway couldn't just bypass it with a basic redirect that doesn't even understand the protocol.

**Comment by lidel** [comment]
Good idea @schomatis, however I've filled https://github.com/ipfs/public-gateway-checker/issues/157 for tracking that – dynamic CID complicate the app by an order of magnitude, and need to be tackled in separate PR, and I want to land this before that, as its already better than old state.
